frontend/src/Frontend/Foundation.hs

```diff
@@ -96,13 +98,11 @@ type family ReflexValue (f :: * -> *) x where
 
     ReflexValue (Event t) x = Event t x
 
-addDecimalToString :: Text -> Text
-addDecimalToString s =
-    case T.find f s of
-      Nothing -> s <> ".0"
-      Just _ -> s
+appendDecimalToText :: Text -> Text
+appendDecimalToText t = case readP_to_S scientificP s of 
+    [] -> T.pack s
+    [(x,_)] -> tshow x
+    [_,(x,_)] -> tshow x
```

Comment:
This case is not quite 1-1 with what you had. It will convert `"3e-7"` into `"3.0e-7"`. If we don't want that, then we can squash the `[]` case together with this one, and drop the error message: 


```haskell
appendDecimalToText :: Text -> Text
appendDecimalToText t = case readP_to_S scientificP s of 
    [(x,_)] -> tshow x
    _ -> T.unpack t
```

Reply:
We also don't need to impossible case: 

```haskell
appendDecimalToText :: Text -> Text
appendDecimalToText t = case readP_to_S scientificP s of 
    [(x,_)] -> tshow x
    [_,(x,_)] -> tshow x
    _ -> T.unpack t
```

Reply:
Fixed